test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+      const expectedClearedBalance: number = 0
+      const expectedUnclearedBalance: number = 0;
+      const expectedRevocationCounter: number = 1;
+
+      async function checkRevocation(wasPaymentCleared: boolean) {
+        await checkCardPaymentProcessorState([payment]);
+        await expect(cardPaymentProcessor.connect(admin).revokePayment(
+          authorizationId,
+          reversingPaymentCorrelationId,
+          parentTxHash
+        )).to.changeTokenBalances(
+          brlcMock,
+          [cardPaymentProcessor, payment.account],
+          [-payment.account, +payment.amount]
+        ).and.to.emit(
+          cardPaymentProcessor,
+          "RevokePayment"
+        ).withArgs(
+          authorizationId,
+          reversingPaymentCorrelationId,
+          payment.account.address,
+          payment.amount,
+          expectedClearedBalance,
+          expectedUnclearedBalance,
+          wasPaymentCleared,
+          parentTxHash,
+          expectedRevocationCounter
+        );
+        payment.status = PaymentStatus.Revoked;
+        payment.revocationCounter = expectedRevocationCounter;
+        await checkCardPaymentProcessorState([payment]);
+      }
+
+      it("If payment is uncleared", async () => {
+        const wasPaymentCleared: boolean = false;
+        await checkRevocation(wasPaymentCleared);
+      });
+
+      it("If payment is cleared", async () => {
+        const wasPaymentCleared: boolean = true;
+        await clearPayments([payment], admin);
+        await checkRevocation(wasPaymentCleared);
+      });
+    })
+  });
+
+  describe("Function 'reversePayment()'", async () => {
+    const reversingPaymentCorrelationId: string = SOME_BYTES16_STRING;
+    const parentTxHash: string = SOME_BYTES32_STRING;
+
+    let payment: TestPayment;
+    let authorizationId: string;
+    let admin: SignerWithAddress;
+
+    beforeEach(async () => {
+      payment = {
+        authorizationId: 876,
+        account: user1,
+        amount: 765,
+        status: PaymentStatus.Nonexistent,
+        makingPaymentCorrelationId: 543,
+        parentTxHash: parentTxHash
+      };
+      admin = user2;
+      authorizationId = createBytesString(payment.authorizationId, BYTES16_LENGTH);
+      await setUpContractsForPayments([payment]);
+      await makePayments([payment]);
+      await addAccountToCardPaymentProcessorWhitelist(admin);
+    });
+
+    it("Is reverted if the contract is paused", async () => {
+      await proveTx(cardPaymentProcessor.setPauser(deployer.address));
+      await proveTx(cardPaymentProcessor.pause());
+      await expect(cardPaymentProcessor.connect(admin).reversePayment(
+        authorizationId,
+        reversingPaymentCorrelationId,
+        parentTxHash
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_CONTRACT_IS_PAUSED);
+    });
+
+    it("Is reverted if the caller is not whitelisted", async () => {
+      await expect(cardPaymentProcessor.connect(deployer).reversePayment(
```

Comment:
Codacy has a fix for the issue: [Replace `cardPaymentProcessor.connect(deployer)` with `⏎········cardPaymentProcessor⏎··········.connect(deployer)⏎··········`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
      await expect(
```